### PR TITLE
Release 1.2.0

### PR DESCRIPTION
**Pull request by vegaro**
- Bumped iOS to 3.4.0 [Changelog here](https://github.com/RevenueCat/purchases-ios/releases)
- Bumped Android to 3.2.0 [Changelog here](https://github.com/RevenueCat/purchases-android/releases)
- Upgrade Flutter Android APIs (https://github.com/RevenueCat/purchases-flutter/pull/74)
- Adds userDefaultsSuiteName as an option when setting up the SDK (https://github.com/RevenueCat/purchases-flutter/pull/70)
- Added managementURL to PurchaserInfo (https://github.com/RevenueCat/purchases-flutter/pull/67)
- Added originalPurchaseDate to PurchaserInfo (https://github.com/RevenueCat/purchases-flutter/pull/66)
- Added `setProxyURL` (https://github.com/RevenueCat/purchases-flutter/pull/63)
- Exposes `setFinishTransactions` (https://github.com/RevenueCat/purchases-flutter/pull/45)
- Adds missing error codes (https://github.com/RevenueCat/purchases-flutter/pull/64)
- Adds new headers for platformFlavor and platformFlavorVersion (https://github.com/RevenueCat/purchases-flutter/pull/57, https://github.com/RevenueCat/purchases-flutter/pull/46)

### CODE REVIEW

**Comment by vegaro** [comment]
@aboedo bump

**Comment by aboedo** [comment]
@vegaro sorry I'd missed this! reviewing

File: lib/errors.dart [hunk 1]
```diff
@@ -25,8 +25,31 @@ enum PurchasesErrorCode {
...
+  ///        print("User cancelled");
+  ///     } else if (errorCode == PurchasesErrorCode.purchaseNotAllowedError) {
+  ///        print("User not allowed to purchase");
+  ///     }
+  /// }
+  /// ```
   static PurchasesErrorCode getErrorCode(PlatformException e) {
-    return PurchasesErrorCode.values[int.parse(e.code)];
+    var errorCode = int.parse(e.code);
+    if (errorCode >= PurchasesErrorCode.values.length) {
```
[the remark this answers is not in the document]
Comment (vegaro) [reply lib/errors.dart:49]: Yes, I wonder if we could break the compilation somehow, or maybe add a test?

[the remark this answers is not in the document]
Comment (aboedo) [reply lib/errors.dart:49]: I was actually thinking about that same thing for the swift enum in purchases-ios. a test might be the way to go, even if it only covers that the number of possible values is the same on both sides

**Pull request merged**